### PR TITLE
[DOC] Corrected spelling errors

### PR DESCRIPTION
**Pull request by kachayev**
## Types of changes
Various spelling errors have been fixed in the docstrings and examples.

## Motivation and context / Related issue
This pull request includes substantial follow-up to the previous fix #462.

In addition, I corrected the markup in multiple locations, which reduced the number of warnings generated during the documentation build process.

## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [ ] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

File: examples/others/plot_logo.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 In this example we plot the logo of the POT toolbox.
 
 This logo is that it is done 100% in Python and generated using
-matplotlib and ploting teh solution of the EMD solver from POT.
+matplotlib and plotting teh solution of the EMD solver from POT.
```
Comment (rflamary) [inline examples/others/plot_logo.py:11]:
```suggestion
matplotlib and plotting the solution of the EMD solver from POT.
```

**Review by rflamary — APPROVED** [review]
Hello @kachayev and thanks a bunch for this major cleanup of the documentation (we needed it).

I have seen a remaining typo (that is clearly mine) and I would like to ask you to add this PR (and the other one already merged) with a  short description in the RELEASE file so that all your work is clearly visible and noted there (we do that for all merge anyways). After that we can merge it.

**Comment by kachayev** [comment]
@rflamary I pushed changes you've asked for. I also had a hope to fix issues with reference labels (there are a lot of references that could not be resolved by sphinx-build). I guess it could be done in another PR, as I still haven't been able to find the root cause of the problem (maybe it's because footnotes/references don't have unique ref numbers, which would require major overhaul of how reference papers are numbered).

**Comment by rflamary** [comment]
Yes references are a pain. My initial idea was to keep the same numbers in the docstring and the readme/first page of the doc in order to be consistent but it raises tons of warnings... If you find a solution i would be wonderfull but it can ided wait for another PR your already have many things in this one. I will merge it shortly

**Pull request merged**